add STAC catalog functionality

tomkralidis opened the pull request:
## Overview

This PR implements STAC Catalog functionality (#221) into pygeoapi.  Basic setup:

- this implements STAC **catalog** (STAC API is ongoing in the [stac branch](https://github.com/geopython/pygeoapi/tree/stac))
- STAC catalog is implemented through a pygeoapi FileSystem provider
- configuration connects to a directory tree of geospatial data files (raster or vector) on disk.  Example provider configuration snippet:

```yaml
        provider:
            name: FileSystem
            data: /Users/tomkralidis/Dev/data/gdps
            file_types:
                - .grib2
```
- JSON and HTML responses are supported
- STAC functionality gets triggered via the `/stac` endpoint and binds to any `FileSystemProvider` datasets in the config
- tests in `tests/test_filesystem_provider.py`
- docs will be added to the [docs branch](https://github.com/geopython/pygeoapi/tree/issue-277-documentation)
- a generic breadcrumbs function was added in `pygeoapi.util:create_breadcrumbs` and used as a jinja2 filter.  This was implemented because STAC pathing is arbitirary (vs. OGC API's more fixed `/collections/{collectionId}/items/{itemID}` structure).  Might be useful for other templates
- raster based support is limited to the entire file's properties.  Multiband support could be added as future work
- the provider uses rasterio and fiona (and thus adds dependencies (see `pygeoapi.provider.filesystem:_describe_file`).  Comments/thoughts?  Is it worth it at this point to drop down to the GDAL Python API to remove the dependencies on rasterio/fiona?

jorgejesus commented:
For what I understood on the yml config `pygeoapi-config.yml` 
Should have out of the box support for a small dataset/example:
```
        provider:
            name: FileSystem
            data: tests/data
            file_types:
                - .geojson
```

tomkralidis commented:
@jorgejesus good idea.  I added a sample setup to our default config to expose the test data.  In addition:

- fix asset URLs to be absolute
- fix tests
- moved fiona and rasterio into `requirements-provider.txt` so that STAC can still work (albeit without geospatial data description) as required

tomkralidis commented:
FYI test failures are unrelated.
